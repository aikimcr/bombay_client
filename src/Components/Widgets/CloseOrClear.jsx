import PropTypes from "prop-types";

import "./OldLabeledInput.scss";

function CloseOrClear(props) {
  const buttonClass = props.className || "close-or-clear";

  return (
    <button className={buttonClass} onClick={props.onClick}>
      &#x2715;
    </button>
  );
}

CloseOrClear.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
};

export default CloseOrClear;
