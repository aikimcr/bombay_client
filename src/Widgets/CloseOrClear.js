import PropTypes from 'prop-types'
import { useRef } from 'react';

import './LabeledInput.scss';

function CloseOrClear(props) {
    const buttonClass = props.className || 'close-or-clear';

    return (
        <button className={buttonClass} onClick={props.onClick}>&#x2715;</button>
    )
}

CloseOrClear.propTypes = {
    onClick: PropTypes.func.isRequired,
    className: PropTypes.string,
}

export default CloseOrClear