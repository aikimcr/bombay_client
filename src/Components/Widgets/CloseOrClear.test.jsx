import { act, render, screen } from '@testing-library/react';

import CloseOrClear from './CloseOrClear.jsx';

jest.useFakeTimers();

it('should show a simple labeled input', async () => {
  const clickHandler = jest.fn();

  const { asFragment } = render(<CloseOrClear onClick={clickHandler} />);

  expect(asFragment).toMatchSnapshot();
});

it('should call the click handler', async () => {
  const clickHandler = jest.fn();

  render(<CloseOrClear onClick={clickHandler} />);

  expect(clickHandler).not.toBeCalled();

  const button = screen.getByText('\u2715');
  button.click();

  expect(clickHandler).toBeCalledTimes(1);
});
