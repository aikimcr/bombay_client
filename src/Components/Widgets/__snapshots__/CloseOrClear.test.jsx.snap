// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show a simple labeled input 1`] = `[Function]`;
